src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/CourtService.kt

```diff
@@ -236,6 +242,33 @@ class CourtService(
       )
     }
 
+  private fun validateVideoLinkBookingSpecification(specification: VideoLinkBookingSpecification) {
+    validateVideoLinkAppointment(specification.main, "Main")
+    specification.pre?.let { validateVideoLinkAppointment(it, "Pre") }
+    specification.post?.let { validateVideoLinkAppointment(it, "Post") }
+  }
+
+  private fun validateVideoLinkBookingUpdateSpecification(specification: VideoLinkBookingUpdateSpecification) {
+    validateVideoLinkAppointment(specification.main, "Main")
+    specification.pre?.let { validateVideoLinkAppointment(it, "Pre") }
+    specification.post?.let { validateVideoLinkAppointment(it, "Post") }
+  }
+
+  private fun validateVideoLinkAppointment(appointment: VideoLinkAppointmentSpecification, prefix: String) {
+    if (appointment.startTime.isBefore(LocalDateTime.now(clock))) {
+      throw ValidationException("$prefix appointment start time must be in the future.")
+    }
+    if (!appointment.startTime.isBefore(appointment.endTime)) {
+      throw ValidationException("$prefix appointment start time must precede end time.")
+    }
+    appointment.locationId?.let {
+      val location = prisonApiService.getLocation(it) ?: throw ValidationException("$prefix locationId $it not found in NOMIS.")
```

Comment:
meh... would be slightly more verbose but could potentially get locations for an agency - would be a single call rather than upto 3 and you could also check all locations existed in the same prison? 🤷

Reply:
6 vs 12/2?  The individual locationDtos have an 'agencyId' field in the response so I could ensure that they all had the same agencyId...  That's sort of contrary to removing the location type check above though.  Should I put business logic in whereabouts-api (beyond safety checks) or leave that to BVL and prisonstaffhub?

Reply:
There are 85 locations of type 'APP' at WWI. 16 of those have locationType == 'VIDE'

Reply:
To look up locations by agencyId I need the agencyId.  I can either hard-wire that to 'WWI' or ask prison-api for offender details for the bookingId and take the offender's agencyId - or the client has to send the agencyId in the create or update booking request.

Reply:
Decided to stay with individual look-ups for now.  Can change if you think it is important though...